packages/core-p2p/lib/server/versions/1/handlers.js

```diff
@@ -359,7 +359,15 @@ exports.getBlocks = {
    */
   async handler (request, h) {
     try {
-      const blocks = await container.resolvePlugin('database').getBlocks(parseInt(request.query.lastBlockHeight) + 1, 400)
+      let database = container.resolvePlugin('database')
+      let reqBlockHeight = parseInt(request.query.lastBlockHeight)
+      let blocks = []
+      if (!request.query.lastBlockHeight || Number.isNaN(reqBlockHeight)) {
+        blocks.push(await database.getLastBlock())
```

Comment:
You should avoid calling the database as the last block there could be wrong as data is periodically stored.

`blocks.push(blockchain.getLastBlock())`

Reply:
`My intention was to return the last block at the time of API call.` then you have to use `blocks.push(blockchain.getLastBlock())` as I said in my comment because the database will have older data. You should always refer to in-memory state for `getLast*` data.

Reply:
Ah I see, yes sir I'd prefer that approach too.